### PR TITLE
Implement fancy placeholder focused data-flow viz

### PR DESCRIPTION
**Pull request by kaushikcfd**
```python
A = pt.make_placeholder("A", shape=(10, 4), dtype=np.float64)
x1 = pt.make_placeholder("x1", shape=4, dtype=np.float64)
x2 = pt.make_placeholder("x2", shape=4, dtype=np.float64)


y = A @ (2*x1 + 3*x2)

pt.show_fancy_placeholder_data_flow(y)
```

plots the graph:

### CODE REVIEW

**Comment by kaushikcfd** [comment]
Tried visualizing the wave operator from `inducer/grudge`. Looks like:

image

File: pytato/utils.py [hunk 1]
```diff
@@ -596,4 +596,25 @@ def get_common_dtype_of_ary_or_scalars(ary_or_scalars: Sequence[ArrayOrScalar]
     return np.find_common_type(array_types=array_types,
                                scalar_types=scalar_types)
 
+
+def get_einsum_subscript_str(expr: Einsum) -> str:
```
Comment (inducer) [inline pytato/utils.py:600]: Add a short docstring?

**Review by inducer — COMMENTED** [review]
LGTM, just a few housekeeping things. If you don't feel you have the time, say so, and I can try and take care of them.

File: pytato/visualization/fancy_placeholder_data_flow.py
[the remark this answers is not in the document]
Comment (kaushikcfd) [reply pytato/visualization/fancy_placeholder_data_flow.py:1]: Done! Just checked if the control flow returns without error. Actually verifying the graphviz plot is left for later.

File: pytato/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaushikcfd) [reply pytato/utils.py:601]: Thanks done!

**Comment by inducer** [comment]
Thx!

**Comment by inducer** [comment]
Generalized the test a bit, added a graphviz dep. I think the arraycontext failure was down to flakiness, but I don't have time to dig right now. Filed as https://github.com/inducer/arraycontext/issues/233.

**Pull request merged**